libexec/swiftenv-install

```diff
@@ -193,21 +193,44 @@ build_version() {
 }
 
 find_binary_url_from_api() {
-  vlog "Checking for a URL for the $VERSION on $(get_platform)."
-
-  status_code=$(curl -o /dev/null --silent --head --write-out '%{http_code}' "https://swiftenv-api.fuller.li/versions/$VERSION/binaries/$(get_platform)" || true)
+  vlog "Checking for a URL for Swift $VERSION on $(get_platform)."
+
+  case $(get_platform) in
+    'Ubuntu:16.04' )
+      SWIFT_BUILD_FOLDER="ubuntu1604"
+      SWIFT_ZIP_FILE="ubuntu16.04"
+      ;;
+    'Ubuntu:18.04' )
+      SWIFT_BUILD_FOLDER="ubuntu1804"
+      SWIFT_ZIP_FILE="ubuntu18.04"
+      ;;
+    'Ubuntu:20.04' )
+      SWIFT_BUILD_FOLDER="ubuntu2004"
+      SWIFT_ZIP_FILE="ubuntu20.04"
+      ;;
+    'CentOS Linux:7')
+      SWIFT_BUILD_FOLDER="centos7"
+      SWIFT_ZIP_FILE="centos7"
+      ;;
+    'CentOS Linux:8')
+      SWIFT_BUILD_FOLDER="centos8"
+      SWIFT_ZIP_FILE="centos8"
+      ;;
+    'Amazon Linux:2')
+      SWIFT_BUILD_FOLDER="amazonlinux2"
+      SWIFT_ZIP_FILE="amazonlinux2"
+      ;;
+    'osx')
+      url="https://swift.org/builds/swift-$VERSION-release/xcode/swift-$VERSION-RELEASE/swift-$VERSION-RELEASE-osx.pkg"
+      return
+      ;;
+    *)
+      echo "There was a problem identifying your OS $(get_platform)"
+      exit 1
+      ;;
+  esac
 
-  if [ "$status_code" = "404" ]; then
-    vlog "Did not find a binary release for $VERSION on $(get_platform)."
-  elif [ "$status_code" = "200" ]; then
-    url="$(curl --silent https://swiftenv-api.fuller.li/versions/$VERSION/binaries/$(get_platform) -H 'Accept: text/plain')"
-  elif [ "$status_code" = "000" ]; then
-    echo "There was a problem checking for a binary release of $VERSION, could not connect to the swiftenv API."
-    exit 1
-  else
-    echo "There was a problem checking for a binary release of $VERSION, server returned $status_code."
-    exit 1
-  fi
+  url="https://swift.org/builds/swift-$VERSION-release/$SWIFT_BUILD_FOLDER/swift-$VERSION-RELEASE/swift-$VERSION-RELEASE-$SWIFT_ZIP_FILE.tar.gz"
```

Comment:
This appears to break installing Swift snapshots, the URL format differs, before:

```
$ env SWIFTENV_PLATFORM=ubuntu20.04 ./bin/swiftenv install DEVELOPMENT-SNAPSHOT-2021-07-24-a
Downloading https://swift.org/builds/development/ubuntu2004/swift-DEVELOPMENT-SNAPSHOT-2021-07-24-a/swift-DEVELOPMENT-SNAPSHOT-2021-07-24-a-ubuntu20.04.tar.gz
```

Now:

```
$ set -x SWIFT_ZIP_FILE ubuntu20.04
$ set -x SWIFT_BUILD_FOLDER ubuntu2004
$ set -x VERSION DEVELOPMENT-SNAPSHOT-2021-07-24-a
$ set -x URL "https://swift.org/builds/swift-$VERSION-release/$SWIFT_BUILD_FOLDER/swift-$VERSION-RELEASE/swift-$VERSION-RELEASE-$SWIFT_ZIP_FILE.tar.gz"

$ echo $URL
https://swift.org/builds/swift-DEVELOPMENT-SNAPSHOT-2021-07-24-a-release/ubuntu2004/swift-DEVELOPMENT-SNAPSHOT-2021-07-24-a-RELEASE/swift-DEVELOPMENT-SNAPSHOT-2021-07-24-a-RELEASE-ubuntu20.04.tar.gz

$ curl -I $URL
HTTP/1.1 404 Not Found
```

Reply:
I'll fix. Out of interest how are you getting the snapshot list?

Reply:
The API scrapes the downloads page (any anchor tag with href's containing `.tar.gz` or `.pkg` extensions) to index the data (https://github.com/kylef/swiftenv-api/blob/master/extract.py#L53).

Perhaps here we could use a check if "SNAPSHOT" is in the name and fallback to different approach (or even approach with using the API as we have it today). That would push the problem out of this PR to be addressed later.

It looks like there has been different formats with branch names, development branches etc:

```
osx: https://swift.org/builds/swift-3.1-branch/xcode/swift-3.1-DEVELOPMENT-SNAPSHOT-2017-02-09-a/swift-3.1-DEVELOPMENT-SNAPSHOT-2017-02-09-a-osx.pkg
amazonlinux2: https://swift.org/builds/development/amazonlinux2/swift-DEVELOPMENT-SNAPSHOT-2021-08-17-a/swift-DEVELOPMENT-SNAPSHOT-2021-08-17-a-amazonlinux2.tar.gz
```

Reply:
All the binary URLs per Swift major versions can be found in https://github.com/kylef/swiftenv-api/tree/master/versions . There's two types of snapshots (unversioned snapshots and version branch snapshots within major directories). Such as 5.5's development snapshots: https://github.com/kylef/swiftenv-api/blob/master/versions/5/5.5-DEVELOPMENT-SNAPSHOT/2021-04-19-a.yaml

Reply:
This is fixed for both development build and branch development builds